pkg/cli/cmd/credential/register/azure/wi/workloadidentity_test.go

```diff
@@ -0,0 +1,233 @@
+limitations under the License.
+*/
+
+package wi
+
+import (
+	"context"
+	"fmt"
+	"path"
+	"testing"
+
+	"go.uber.org/mock/gomock"
+
+	v1 "github.com/radius-project/radius/pkg/armrpc/api/v1"
+	"github.com/radius-project/radius/pkg/cli"
+	"github.com/radius-project/radius/pkg/cli/cmd/credential/common"
+	"github.com/radius-project/radius/pkg/cli/connections"
+	cli_credential "github.com/radius-project/radius/pkg/cli/credential"
+	"github.com/radius-project/radius/pkg/cli/framework"
+	"github.com/radius-project/radius/pkg/cli/output"
+	"github.com/radius-project/radius/pkg/cli/workspaces"
+	"github.com/radius-project/radius/pkg/to"
+	ucp "github.com/radius-project/radius/pkg/ucp/api/v20231001preview"
+	"github.com/radius-project/radius/test/radcli"
+	"github.com/stretchr/testify/require"
+	"gopkg.in/yaml.v3"
+)
+
+func Test_CommandValidation(t *testing.T) {
+	radcli.SharedCommandValidation(t, NewCommand)
+}
+
+func Test_Validate(t *testing.T) {
+	configWithWorkspace := radcli.LoadConfigWithWorkspace(t)
+	testcases := []radcli.ValidateInput{
+		{
+			Name: "Valid Azure command",
+			Input: []string{
+				"--client-id", "abcd",
+				"--tenant-id", "ijkl",
+			},
+			ExpectedValid: true,
+			ConfigHolder:  framework.ConfigHolder{Config: configWithWorkspace},
+		},
+		{
+			Name: "Azure command with fallback workspace",
+			Input: []string{
+				"--client-id", "abcd",
+				"--tenant-id", "ijkl",
+			},
+			ExpectedValid: true,
+			ConfigHolder:  framework.ConfigHolder{Config: radcli.LoadEmptyConfig(t)},
+		},
+		{
+			Name: "Azure command with too many positional args",
+			Input: []string{
+				"letsgoooooo",
+				"--client-id", "abcd",
+				"--tenant-id", "ijkl",
+			},
+			ExpectedValid: false,
+			ConfigHolder:  framework.ConfigHolder{Config: configWithWorkspace},
+		},
+		{
+			Name: "Azure command without client-id",
+			Input: []string{
+				"--tenant-id", "ijkl",
+			},
+			ExpectedValid: false,
+			ConfigHolder:  framework.ConfigHolder{Config: configWithWorkspace},
+		},
+		{
+			Name: "Azure command without tenant-id",
+			Input: []string{
+				"--client-id", "abcd",
+			},
+			ExpectedValid: false,
+			ConfigHolder:  framework.ConfigHolder{Config: configWithWorkspace},
+		},
+	}
+	radcli.SharedValidateValidation(t, NewCommand, testcases)
+}
+
+func Test_Run(t *testing.T) {
+	t.Run("Create azure provider", func(t *testing.T) {
+		t.Run("Success", func(t *testing.T) {
+			ctrl := gomock.NewController(t)
+
+			// We need to isolate the configuration because we're going to make edits
+			configPath := path.Join(t.TempDir(), "config.yaml")
+
+			yamlData, err := yaml.Marshal(map[string]any{
+				"workspaces": cli.WorkspaceSection{
+					Default: "a",
+					Items: map[string]workspaces.Workspace{
+						"a": {
+							Connection: map[string]any{
+								"kind":    workspaces.KindKubernetes,
+								"context": "my-context",
+							},
+							Source: workspaces.SourceUserConfig,
+
+							// Will have provider info added
```

Comment:
Is this to be addressed in a different PR?

Reply:
I copied the code from the existing `rad credential register azure` code, so it looks like this hasn't been added yet.

Reply:
@willdavsmith I know this is a copy of existing code.. but while you are at it can you move this comment into an issue, if this is still something in our plan? Leaving the comment here isn't very helpful.

Reply:
Talked to ryan - looks like this test was out of date. provider info should not be added to the workspace. I have cleaned up this test